java/ql/lib/semmle/code/java/security/performance/ReDoSUtilSpecific.qll

```diff
@@ -0,0 +1,75 @@
+/**
+ * This module should provide a class hierarchy corresponding to a parse tree of regular expressions.
+ * This is the interface to the shared ReDoS library.
+ */
+
+import java
```

Comment:
This qll defines the input to a cross-language shared module, so the `import java` ought to be private.
```suggestion
private import java
```